### PR TITLE
Show member detail page on id base

### PR DESCRIPTION
**Pull request by ghulamhafeez**
### Implementation

- Display member detail page based on ID with actual data
- Created date formatter function for get member age.
- Update detail page design

### Resources
https://github.com/user-attachments/assets/5110a655-7c35-4061-8e58-fbda511fd22d

### CODE REVIEW

File: src/app/members/components/MemberDetails.tsx [hunk 1]
```diff
@@ -11,117 +11,118 @@ import {
...
 import { MemberRole } from "@/constant/constant";
+import { getAge } from "@/utils/ageFormater";
 
 export const MemberDetails = ({ userInfo }: { userInfo: UserDetails }) => {
   return (
-    <div>
-      <Card>
-        <CardHeader className="flex flex-col justify-between md:flex-row">
-          <div className="flex flex-row items-center space-x-4">
+    <Card>
```
Comment (Mubashar1009) [inline src/app/members/components/MemberDetails.tsx:18]: Nice

File: src/app/members/components/MemberDetails.tsx [hunk 2]
```diff
@@ -11,117 +11,118 @@ import {
...
+            <h3 className="flex items-center gap-2 text-sm font-semibold">
+              <div className="flex items-center justify-center w-10 h-10 rounded-full bg-gray-100">
+                <UserIcon className="w-4 h-4" />
               </div>
+              <span>Personal Information</span>
+            </h3>
+            <div className="pl-12 space-y-2 ">
+              <p className="text-xs">
+                <span className="font-medium text-muted-foreground">Age:</span>{" "}
+                {getAge(userInfo.date_of_birth)} years
```
Comment (Mubashar1009) [inline src/app/members/components/MemberDetails.tsx:52]: can we use it outside `p` tag?

File: src/app/members/components/MemberDetails.tsx [hunk 3]
```diff
@@ -11,117 +11,118 @@ import {
...
+              </p>
+            </div>
+            <div className="pl-12 space-y-2">
+              <p className="text-xs">
+                <span className="font-medium text-muted-foreground">Role:</span>{" "}
```
Comment (Mubashar1009) [inline src/app/members/components/MemberDetails.tsx:57]: remove inverted commas actually it add a space

File: src/app/members/details/[id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    id: string;
+  };
+};
+
+const Page = async ({ params }: Parameters) => {
+  const { id } = params;
+  const userInfo = await getUserById(id);
+  return (
+    <div className="p-8">
+     <FormWrapper><MemberDetails userInfo={userInfo} /></FormWrapper>
```
Comment (Mubashar1009) [inline src/app/members/details/[id]/page.tsx:17]: Why we need form wrapper here ?

File: src/app/members/details/[id]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import { getUserById } from "../../actions/get-user-by-id";
+import FormWrapper from "@/common/FormWrapper";
+
+type Parameters = {
+  params: {
+    id: string;
+  };
+};
+
+const Page = async ({ params }: Parameters) => {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/members/details/[id]/page.tsx:12]: why we can't use `useParams`

File: src/app/members/details/[id]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { MemberDetails } from "../../components/MemberDetails";
+import { getUserById } from "../../actions/get-user-by-id";
+import FormWrapper from "@/common/FormWrapper";
+
+type Parameters = {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/members/details/[id]/page.tsx:6]: why you using object inside a object , can simplify it

File: src/app/members/components/MemberDetails.tsx (hunk 1 above)
Comment (ghulamhafeez) [reply src/app/members/components/MemberDetails.tsx:18]: Fixed

File: src/app/members/components/MemberDetails.tsx (hunk 2 above)
Comment (ghulamhafeez) [reply src/app/members/components/MemberDetails.tsx:52]: I think it's fine if i move it outside the <p> tag it disturbed the UI.

File: src/app/members/details/[id]/page.tsx (hunk 5 above)
Comment (ghulamhafeez) [reply src/app/members/details/[id]/page.tsx:12]: Both ways are fine

File: src/app/members/details/[id]/page.tsx (hunk 4 above)
Comment (ghulamhafeez) [reply src/app/members/details/[id]/page.tsx:17]: Because this detail page has a similar layout as the form.

**Review by Mubashar1009 — APPROVED** [review]

**Pull request merged**